nemo/export/trt_llm/tensorrt_llm_run.py

```diff
@@ -210,60 +229,84 @@ def _forward(
+        #     line_encoded = torch.nested.to_padded_tensor(
+        #         torch.nested.nested_tensor(input_tensors, dtype=torch.int32), pad_id
+        #     ).cuda()
 
-        input_lengths = torch.tensor(input_lengths, dtype=torch.int32).cuda()
+        # input_lengths = torch.tensor(input_lengths, dtype=torch.int32).cuda()
 
-        if prompt_table is None:
-            ptuning_args = []
-        else:
-            if task_vocab_size is None:
-                raise Exception("task_vocab_size cannot be None")
+        # if prompt_table is None:
+        #     ptuning_args = []
+        # else:
+        #     if task_vocab_size is None:
+        #         raise Exception("task_vocab_size cannot be None")
 
-            task_vocab_size = torch.tensor([task_vocab_size], dtype=torch.int32, device="cuda")
-            task_ids = torch.tensor(task_ids, dtype=torch.int32, device="cuda")
-            prompt_table = prompt_table.cuda()
-            ptuning_args = [prompt_table, task_ids, task_vocab_size]
+        #     task_vocab_size = torch.tensor([task_vocab_size], dtype=torch.int32, device="cuda")
+        #     task_ids = torch.tensor(task_ids, dtype=torch.int32, device="cuda")
+        #     prompt_table = prompt_table.cuda()
+        #     ptuning_args = [prompt_table, task_ids, task_vocab_size]
 
         with torch.no_grad():
-            sampling_config.top_k = top_k
-            sampling_config.top_p = top_p
-            sampling_config.temperature = temperature
-            for key, param in sampling_kwargs.items():
-                # set any additional SamplingConfig kwargs
-                setattr(sampling_config, key, param)
-
-            decoder.setup(
-                batch_size,
-                max_context_length=max_length,
+            # sampling_config.top_k = top_k
+            # sampling_config.top_p = top_p
+            # sampling_config.temperature = temperature
+            # for key, param in sampling_kwargs.items():
+            #     # set any additional SamplingConfig kwargs
+            #     setattr(sampling_config, key, param)
+
+            # decoder.setup(
+            #     batch_size,
+            #     max_context_length=max_length,
+            #     max_new_tokens=max_output_len,
+            #     lora_manager=lora_manager,
+            #     lora_uids=lora_uids,
+            # )
+
+            # outputs = decoder.decode(
+            #     line_encoded,
+            #     input_lengths,
+            #     sampling_config,
+            #     *ptuning_args,
+            #     stop_words_list=stop_words_list,
+            #     bad_words_list=bad_words_list,
+            #     no_repeat_ngram_size=no_repeat_ngram_size,
+            #     streaming=streaming,
+            #     output_sequence_lengths=True,
+            #     return_dict=True,
+            # )
+
+            prompt_tasks = None if task_ids is None else ",".join(str(task) for task in task_ids)
+
+            outputs = decoder.generate(
```

Comment:
I think you can just use the HLAPI like LLM to run inference. The API is much cleaner to call than this ModelRunnerCpp